learn/conversation-draft-cache.ts

```diff
@@ -0,0 +1,112 @@
+import { Cache } from './cache';
+import { getDrafts, saveDrafts } from '../electron/filesystem';
+import core from '../chat/core';
+
+export interface ConversationCachedMessage {
+  channel: string;
+  message: string;
+}
+
+export class ConversationDraftRecord {
+  channel: string;
+  message: string;
+
+  constructor(channel: string, message?: string) {
+    this.channel = channel;
+    this.message = message || '';
+  }
+}
+
+export class ConversationDraftCache extends Cache<ConversationDraftRecord> {
+  private lastCacheSave = Date.now();
+  private cacheAlreadyLoaded = false;
+  private useCache = true;
+  private diskSaveTimerInSeconds = 60;
+  private diskSaveInterval: NodeJS.Timer | null = null;
+  private currentlyCachedCharacter = '';
+  private resetListenerActive = false;
+
+  async loadCache(): Promise<void> {
+    if (!this.resetListenerActive) {
+      core.connection.onEvent('connected', () => {
+        this.resetCacheIfNeeded();
+      });
+      this.resetListenerActive = true;
+    }
+
+    if (!core.connection.character || this.cacheAlreadyLoaded) return;
+
+    const settings = await core.settingsStore.get('settings');
+
+    // Check for opt-out setting on cache.
+    if (settings?.horizonCacheDraftMessages === false) this.useCache = false;
+    this.diskSaveTimerInSeconds =
+      settings?.horizonSaveDraftMessagesToDiskTimer ||
+      this.diskSaveTimerInSeconds;
+
+    if (!this.useCache) {
+      this.cacheAlreadyLoaded = true;
+      return;
+    }
+
+    // Check possible accidents from the user (mostly a backup check in case the setting somehow gets set to e.g. 0)
+    if (this.diskSaveTimerInSeconds < 5) this.diskSaveTimerInSeconds = 5;
```

Comment:
It's a raw number in the settings menu too, but I'm fine w/ that for now (unless we want to go and make a generic ``consts/settings.ts`` file or something). But for now it'd be a tiny bit cleaner if this 5 were a const in this specific file too.